changelog/pending/20231207--cli-new--adds-pulumi-new-ai-functionality.yaml

```diff
@@ -0,0 +1,13 @@
+changes:
+- type: feat
+  scope: cli/new
+  description: |
+  Adds pulumi new --ai functionality.
+  The `pulumi new` command now defaults to a new experience - users can select `ai` or `template` if no arguments are specified.
+
+  If `ai` is selected, we offer a guided experience to iteratively prompt Pulumi AI to craft a program for you.
+
+  Additionally, for a less interactive experience, we offer the flags `--ai` and `--language`. If `--ai <prompt>` is provided,
+  we will begin the interactive prompting session with the provided prompt. If `--language <language>` is provided with a supported language,
+  we will skip the language selection prompt and begin the interactive prompting session with the provided language. Addionally, if `--yes` is provided,
+  we will skip the verification step and immediately create the project generated by the given prompt and language.
```

Comment:
Would be great to add some of this same information into the docs for `pulumi new` as well (in pkg/cmd/pulumi/new.go).